server.js

```diff
@@ -1,27 +1,85 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import listEndpoints from "express-list-endpoints";
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+//Schema allows us to reuse code
+//We can combine a schema for certain propertys that shall remain together with properties that shall vary
+//enum will add the only acceptable names to be able to pass to the name property
+//trim deletes whitespace from beginning and end of string - not in between
+//new Date needs to have anonymus function in order not to execute when page loads
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    minlength: 5,
+    maxlength: 140,
+    trim: true
+  },
```

Comment:
Nice with the trim property also! One property that could be added as well is the required, like this: 
`const HappyThoughtSchema = new mongoose.Schema({
  message: {
    type: String,
    required: true,
    minlength: 5,
    maxlength: 140
  },`
  
  In the brief for the project they wrote about it in the "The thought model" section in Project to build: 
  "- `message` - the text of the thought
    → Required
    → Min length of 5 characters
    → Max length of 140 characters